File: marking/app/src/main/java/com/htchan/marking/baseModel/Item.java
package com.htchan.marking.baseModel;

import android.content.ContentValues;
import android.database.Cursor;

public class Item extends AbstractItem{
    public static final String TABLE_NAME = "item";
    public Item(long id) {
        fromCursor(DatabaseHelper.DatabaseHelper().getCursorById(TABLE_NAME, id));
        this.table = TABLE_NAME;
    }
    public Item(String title, String parentTable, long parentId) {
        super(TABLE_NAME, -1, title, parentTable, parentId);
    }
    @Override
    public void save() {
        id = DatabaseHelper.DatabaseHelper().saveRecord(Item.TABLE_NAME, this);
    }
    @Override
    public void update() {
        DatabaseHelper.DatabaseHelper().updateRecord(Item.TABLE_NAME, this);
    }
    @Override
    public void delete() {
        for(AbstractItem child : getChildren(table, id)) {
            child.parentTable = parentTable;
            child.parentId = parentId;
            child.update();
        }
        DatabaseHelper.DatabaseHelper().delete(Item.TABLE_NAME, this);
    }
    @Override
    public void deleteChild() {
        DatabaseHelper.DatabaseHelper().deleteChild(Item.TABLE_NAME, this);
    }
    @Override
    public void fromCursor(Cursor cursor) {
        super.fromCursor(cursor);
    }
    @Override
    public ContentValues toValues() {
        return super.toValues();
    }
    @Override
    public String toString() {
        String output = super.toString();
        if (output.equals("encode error")) {
            return output;
        }
        try{
            return "table=" + encode(TABLE_NAME) + "&" +output;
        } catch (Exception e) {
            return "encode error";
        }
    }
    @Override
    public boolean equals(Object obj) {
        if(! (obj instanceof Item)) {
            return false;
        } else {
            Item item = (Item) obj;
            return this.id == item.id;
        }
    }
}

File: marking/lib/models/item.dart
class Item {
  int parentID, id;
  String content;
  
  Item({
    required this.id,
    required this.parentID,
    required this.content
  });
  Item.from(Map<String, dynamic> map):
  id = map['id'] ?? -1,
  parentID = map['parentID'] ?? -1,
  content = map['content'] ?? '';

  Map<String, dynamic> toMap() =>
    {
      'parentID': parentID,
      'content': content
    };

  Map<String, dynamic> toFullMap() =>
    {
      'id': id,
      'parentID': parentID,
      'content': content
    };

  @override
  String toString() =>
    'Item{id: $id, parentID: $parentID, content: $content}';

  @override
  bool operator==(Object other) {
    return toString() == other.toString();
  }
}